File: LemonTracker/Models/ChannelRow.cs
namespace LemonTracker.Models;

public class ChannelRow
{
    public char? Instrument { get; set; }
    public char? Ornament { get; set; }
    public char? Envelope { get; set; }
    public NoteData? NoteData { get; set; } 
    public char? Volume { get; set; }
    public char? Effect { get; set; }
    public char? EffectParamX { get; set; }
    public char? EffectParamY { get; set; }
    public char? EffectParamZ { get; set; }
}

File: src/models/channel-row.ts
import NoteData, { Note } from "./note-data";

export default class ChannelRow {
  instrument: number = 0;
  ornament: number = 0;
  envelope: number = 0;
  noteData: NoteData = new NoteData(Note.None, 0);
  volume: number = 0;
  effect: number = 0;
  effectParamX: number = 0;
  effectParamY: number = 0;
  effectParamZ: number = 0;
}
